### PR TITLE
ci: check conventional commits and auto release

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

**Review by jyeshe — COMMENTED** [review]
worked for me also with scopes 👍 may we have also a pre-commit hook? (checking the commit message and we could take a ride (: and enable mix format --check-formated and credo in the future after refactoring)
EDIT: after convo, this will be optional per contributor.

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+on:
+  push:
+    branches:
+      - main
```
Comment (wj2016) [inline .github/workflows/release-please.yml:4]: The default branch should be `master`?

Comment (jyeshe) [reply .github/workflows/release-please.yml:4]: Good catch and pushing a tag.

Comment (thepiwo) [reply .github/workflows/release-please.yml:4]: @jyeshe the tool will automatically tag and release

**Comment by thepiwo** [comment]
@jyeshe @sborrazas can you approve?

**Review by sborrazas — APPROVED** [review]

**Review by jyeshe — APPROVED** [review]

**Pull request merged**